x-pack/plugins/ml/public/components/annotations_table/annotations_table.js

```diff
@@ -0,0 +1,342 @@
+ */
+
+import PropTypes from 'prop-types';
+import rison from 'rison-node';
+
+import React, {
+  Component
+} from 'react';
+
+import {
+  EuiBadge,
+  EuiButtonIcon,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiLoadingSpinner,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import {
+  RIGHT_ALIGNMENT,
+} from '@elastic/eui/lib/services';
+
+import { formatDate } from '@elastic/eui/lib/services/format';
+import chrome from 'ui/chrome';
+
+import { addItemToRecentlyAccessed } from 'plugins/ml/util/recently_accessed';
+import { ml } from 'plugins/ml/services/ml_api_service';
+import { mlAnomaliesTableService } from '../anomalies_table/anomalies_table_service';
+import { DEFAULT_QUERY_SIZE } from '../../../common/constants/search';
+
+const TIME_FORMAT = 'YYYY-MM-DD HH:mm:ss';
+
+/**
+ * Table component for rendering the lists of annotations for an ML job.
+ */
+class AnnotationsTable extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      annotations: [],
+      isLoading: false
+    };
+  }
+
+  getAnnotations() {
+    const job = this.props.jobs[0];
+    const dataCounts = job.data_counts;
+
+    this.setState({
+      isLoading: true
+    });
+
+    if (dataCounts.processed_record_count > 0) {
+      // Load annotations for the selected job.
+      ml.annotations.getAnnotations({
+        jobIds: [job.job_id],
+        earliestMs: dataCounts.earliest_record_timestamp,
+        latestMs: dataCounts.latest_record_timestamp,
+        maxAnnotations: DEFAULT_QUERY_SIZE
+      }).then((resp) => {
+        this.setState((prevState, props) => ({
+          annotations: resp.annotations[props.jobs[0].job_id] || [],
+          errorMessage: undefined,
+          isLoading: false,
+          jobId: props.jobs[0].job_id
+        }));
+      }).catch((resp) => {
+        console.log('Error loading list of annoations for jobs list:', resp);
+        this.setState({
+          annotations: [],
+          errorMessage: 'Error loading the list of annotations for this job',
+          isLoading: false,
+          jobId: undefined
+        });
+      });
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.annotations === undefined) {
+      this.getAnnotations();
+    }
+  }
+
+  componentWillUpdate() {
+    if (
+      this.props.annotations === undefined &&
+      this.state.isLoading === false &&
+      this.state.jobId !== this.props.jobs[0].job_id
+    ) {
+      this.getAnnotations();
+    }
+  }
+
+  openSingleMetricView(annotation) {
+    // Creates the link to the Single Metric Viewer.
+    // Set the total time range from the start to the end of the annotation,
+    const dataCounts = this.props.jobs[0].data_counts;
+    const from = new Date(dataCounts.earliest_record_timestamp).toISOString();
+    const to = new Date(dataCounts.latest_record_timestamp).toISOString();
+
+    const _g = rison.encode({
```

Comment:
The whole annotations table is based on the forecasts table so that's where I got this code from originally. If there's room for improvement, are you ok that we do that in a follow up for both forecasts and annotations table?

Reply:
Yes fine - do them both in a follow up!

Reply:
As discussed, added a note to https://github.com/elastic/kibana/issues/25856